test/e2e/suite/conformance/certificates/tests.go

```diff
@@ -573,5 +574,62 @@ func (s *Suite) Define() {
 			err = f.Helper().ValidateCertificate(f.Namespace.Name, certName, f.Helper().ValidationSetForUnsupportedFeatureSet(s.UnsupportedFeatures)...)
 			Expect(err).NotTo(HaveOccurred())
 		})
+
+		s.it(f, "should not re-issue a certificate on usages change", func(issuerRef cmmeta.ObjectReference) {
+			testCertificate := &cmapi.Certificate{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "testcert",
+					Namespace: f.Namespace.Name,
+				},
+				Spec: cmapi.CertificateSpec{
+					SecretName: "testcert-tls",
+					IssuerRef:  issuerRef,
+					DNSNames:   []string{s.newDomain()},
+				},
+			}
+			By("Creating a Certificate")
+			err := f.CRClient.Create(ctx, testCertificate)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting for the Certificate to be issued...")
+			err = f.Helper().WaitCertificateIssued(f.Namespace.Name, "testcert", time.Minute*5)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Validating the issued Certificate...")
+			err = f.Helper().ValidateCertificate(f.Namespace.Name, "testcert", f.Helper().ValidationSetForUnsupportedFeatureSet(s.UnsupportedFeatures)...)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Fetching content of the issued Certificate...")
+			sec, err := f.KubeClientSet.CoreV1().Secrets(f.Namespace.Name).
+				Get(context.TODO(), testCertificate.Spec.SecretName, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred(), "failed to get secret containing signed certificate key pair data")
+
+			sec = sec.DeepCopy()
+			crtPEM1 := sec.Data[corev1.TLSCertKey]
+
+			cert, err := f.Helper().CMClient.CertmanagerV1().Certificates(f.Namespace.Name).Get(ctx, "testcert", metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			cert.Spec.Usages = append(cmapi.DefaultKeyUsages(), cmapi.UsageServerAuth)
+			By("Updating usages on Certificate")
+			err = f.CRClient.Update(ctx, cert)
+			Expect(err).NotTo(HaveOccurred())
+			By("Waiting for the Certificate to be ready (if test oks okay it already will be)...")
+			err = f.Helper().WaitCertificateIssued(f.Namespace.Name, "testcert", time.Minute*5)
```

Comment:
I am a bit concerned here that we have a race condition that we are not properly handling (the wait certificate issued gets the certificate before cert-manager has reconciled it and potentially is trying to reissue).

Is there some status that we can check against to make sure that doesn't happen? Perhaps a last transaction time on the readiness condition which is a timestamp after updating the certificate?

Reply:
Would https://github.com/jetstack/cert-manager/issues/1551 help here? also ref https://github.com/jetstack/cert-manager/issues/3585

Reply:
Yep, definitely!

My feeling is that the ready condition would have a different timestamp in this case so is one method of checking, but could be wrong. This addition would actually match the intent better semantically.

Reply:
The timestamp field is `lastTransitionTime`, and its important to note that it records the last time that the condition *transitioned* from a True->False or False->True state. If something was true, and then continues to be true still (even via an update that changes the Message/Reason), the lastTransitionTime will stay the same.

Reply:
Yeah that makes sense.

Its also occurred to me that I don't think this is a correct test anywho- if the usages on the `Certificate` resource has been updated, we _do_ expect a re-issuance. At least, this is the current behaviour. It should be the case where the certificate in the _`Secret`_ has a miss-match of usages.

Reply:
The only time we should trigger a re-issuance is if the current Certificate spec would result in a *CSR* with different parameters to that of the 'CurrentRequest' - if there is no current request, then we fall back to comparing directly against the issued certificate stored in the Secret (however there's a good chance this will mismatch and trigger a re-issuance as CAs can set their own values for these things as-is).

I think that's what you're saying though anyway, so 👍  :)

Reply:
On discussion i moved the test to https://github.com/jetstack/cert-manager/pull/3607 so we can nove on on getting this one merged for the alpha